src/board/mux.go

```diff
@@ -0,0 +1,93 @@
+package board
+
+import (
+	"log"
+
+	"github.com/HyperloopUPV-H8/Backend-H8/board/models"
+	vehicle_models "github.com/HyperloopUPV-H8/Backend-H8/vehicle/models"
+)
+
+const UNRECOGNIZED_CHAN_BUF_SIZE = 100
+
+type BoardMux struct {
+	boards    map[string]models.Board
+	idToBoard map[uint16]string
+
+	sendOrder    func(vehicle_models.Order) error
+	unrecognized chan vehicle_models.Update
+}
+
+func NewMux(config ...MuxConfig) *BoardMux {
+	mux := &BoardMux{
+		unrecognized: make(chan vehicle_models.Update, UNRECOGNIZED_CHAN_BUF_SIZE),
+	}
+
+	for _, conf := range config {
+		conf.UpdateMux(mux)
+	}
+
+	return mux
+}
+
+func (mux *BoardMux) AddBoard(board models.Board) {
+	board.Output(mux.sendOrder)
+	mux.boards[board.Name()] = board
+}
+
+func (mux *BoardMux) AddBoardMapping(idToBoard map[uint16]string) {
```

Comment:
I would call this inside addBoard because if you add a board, you also want the board mapping.

Reply:
Yeah, but maybe you dont have that information when you add the board logic or you have it before. Or maybe you want to change the board logic without changing the mapping. This approach is more flexible